packages/core/plugins/core-plugins-server-internal/src/plugins_system.ts

```diff
@@ -224,21 +224,35 @@ export class PluginsSystem<T extends PluginType> {
 
     this.log.info(`Stopping all plugins.`);
 
-    // Stop plugins in the reverse order of when they were set up.
-    while (this.satupPlugins.length > 0) {
-      const pluginName = this.satupPlugins.pop()!;
+    const reverseDependencyMap = buildReverseDependencyMap(this.plugins);
+    const pluginStopPromiseMap = new Map<PluginName, Promise<void>>();
+    for (let i = this.satupPlugins.length - 1; i > -1; i--) {
+      const pluginName = this.satupPlugins[i];
+      const plugin = this.plugins.get(pluginName)!;
+      const pluginDependant = reverseDependencyMap.get(pluginName)!;
+      const dependantPromises = pluginDependant.map(
+        (dependantName) => pluginStopPromiseMap.get(dependantName)!
+      );
 
-      this.log.debug(`Stopping plugin "${pluginName}"...`);
+      // Stop plugin as soon as all the dependant plugins are stopped.
+      const pluginStopPromise = Promise.all(dependantPromises).then(async () => {
```

Comment:
Old behavior was to throw on first error, so I just kept it tbh. It may be better to wait for everything, but I was kinda scared to change.

Reply:
The only difference is that now we are running multiple stops at the same time. So we may cancel ongoing promises.

IMO, it's OK not to continue calling other `stops`, but I feel uneasy cancelling running calls (despite our _stop is not guaranteed to be called policy_).

Happy to merge as is, though.

Reply:
Fair point. it would be a bit challenging to only cancel the running promises and would over complexify the flow. I will adapt to wait for everything instead and use `allSettled` 👍